src/cryptsetup/cryptsetup-generator.c

```diff
@@ -93,7 +94,21 @@ static int generate_keydev_mount(const char *name, const char *keydev, char **un
                 "[Mount]\n"
                 "What=%s\n"
                 "Where=%s\n"
-                "Options=ro\n", what, where);
+                "Options=ro%s\n", what, where, canfail ? ",nofail" : "");
+
+        if (keydev_timeout) {
+                r = parse_sec_fix_0(keydev_timeout, &timeout_seconds);
+                if (r >= 0) {
+                        r = unit_name_from_path(what, ".device", &device_unit);
+                        r = write_drop_in_format(arg_dest, device_unit, 90, "device-timeout",
+                                "# Automatically generated by systemd-cryptsetup-generator \n\n"
+                                "[Unit]\nJobRunningTimeoutSec=%s", keydev_timeout);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to write device drop-in: %m");
+                } else {
+                        log_warning("Failed to parse timeout for %s, ignoring: %s", where, keydev_timeout);
```

Comment:
use `log_warning_errno(r, …` please, i.e. always pass the original error code along